api/src/com/cloud/network/vpc/OSPFZoneConfig.java

```diff
@@ -0,0 +1,332 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package com.cloud.network.vpc;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.commons.lang.ArrayUtils;
+import org.apache.commons.lang.StringUtils;
+
+import com.cloud.exception.InvalidParameterValueException;
+import com.cloud.utils.net.NetUtils;
+
+public class OSPFZoneConfig {
+
+    private long zoneId;
+    private Protocol protocol;
+    private String ospfArea;
+    private Short helloInterval;
+    private Short deadInterval;
+    private Short retransmitInterval;
+    private Short transitDelay;
```

Comment:
@jburwell lets consider "helloInterval" it is a interval is clear from the name and how it is displayed in UI. Internally it is handled as Short as the MS business logic does not operate on its intrinsic quality of time. Now lets consider that we make it duration as in joda.time pacakge. Now MS business logic need to carry on the overheads of this notion of time by calling the right constructor with additional work required to interpret the parameters and then again doing that conversion when passing this value to virtual router scripts which in turn will dump it in a config file. The tricky part is that this variable conveys notion of time as you rightly put. But the management server business logic need not be aware of this notion of time and can safely transfer the values using Short and thus avoid overheads in conversion back and forth.